### PR TITLE
Fix: search

### PR DESCRIPTION
**Pull request by Janbong**

### CODE REVIEW

File: frontend/src/pages/search.vue [hunk 1]
```diff
@@ -15,146 +15,145 @@
...
+    bookSearchResults.value = itemResults.filter(
+      (item: BaseItemDto) => item.Type === 'Book'
+    );
+
+    personSearchResults.value = (
+      await remote.sdk.newUserApi(getPersonsApi).getPersons({
+        userId: remote.auth.currentUserId.value,
+        searchTerm: searchQuery.value
+      })
+    ).data.Items;
```
Comment (ThibaultNocchi) [inline frontend/src/pages/search.vue:148]:
```suggestion
    ).data.Items || [];
```
Should do the trick :)

File: frontend/src/pages/search.vue [hunk 2]
```diff
@@ -15,146 +15,145 @@
...
+    albumSearchResults.value = itemResults.filter(
+      (item: BaseItemDto) => item.Type === 'MusicAlbum'
+    );
+
+    trackSearchResults.value = itemResults.filter(
+      (item: BaseItemDto) => item.Type === 'Audio'
+    );
+
+    bookSearchResults.value = itemResults.filter(
+      (item: BaseItemDto) => item.Type === 'Book'
```
Comment (ThibaultNocchi) [inline frontend/src/pages/search.vue:140]: You can use the enum for the type checks, such as this:

```suggestion
      (item: BaseItemDto) => item.Type === BaseItemKind.Book
```

**Review by ferferga — CHANGES_REQUESTED** [review]
* We should remove the ``Top results`` tab, because the Virtual Grid is not ready to show cards of different sizes (or we shouldn't use virtual grid at all here, which may even be a more sensible solution if we want to keep that tab, which I'm not a fan of because it's horrible). We need to redesign that page functionally anyway.

* I'd say we should keep the functionality of keeping tabs that don't have any items disabled.

**Review by ferferga — APPROVED** [review]

File: frontend/src/pages/search.vue [hunk 3]
```diff
@@ -1,160 +1,189 @@
...
+);
+
+watch(
+  searchQuery,
+  async () => {
+    if (searchQuery.value !== '') {
+      await performSearchDebounce();
     } else {
-      await this.performSearch();
+      router.back();
```
Comment (ThibaultNocchi) [inline frontend/src/pages/search.vue:120]: What do you think of putting the router.back in the performSearch? That way 1. logic is centralized 2. you can quickly erase your sentence and start typing again without being redirected back to the debounced logic.

Also you may want to use watchDebounced from vueuse?

**Review by ThibaultNocchi — APPROVED** [review]

File: frontend/src/components/Layout/AppBar/SearchField.vue [hunk 4]
```diff
@@ -22,13 +23,23 @@ const searchQuery = computed({
...
-      router.back();
-    } else if (searchQuery.value) {
-      router.replace({ path: '/search', query: { q: value } });
-    } else {
-      router.push({ path: '/search', query: { q: value } });
-    }
+    router.replace({
+      ...router.currentRoute,
+      query: {
+        q: value || undefined
```
Comment (ThibaultNocchi) [inline frontend/src/components/Layout/AppBar/SearchField.vue:29]:
```suggestion
        q: value.trim() || undefined
```

Just an easy QOL change

**Pull request merged**